lib/ws/ws.js

```diff
@@ -9,13 +9,86 @@ var factory = require('./factory');
 var authMethod = 'auth';
 
 /**
+ * race takes an array of promises and is resolved with the first promise
+ * that resolves, or rejects. After the first resolve or rejection, all
+ * future resolutions and rejections will be discarded.
+ * @param {Promise[]} promises
+ * @return {Promise}
+ */
+function race(promises) {
```

Comment:
Also `1) it's pretty heavy filesize-wise` you do realise bluebird is included in the frontend as it's a required dependency of beam-client-node.

I feel like we should remove bluebird as a dependency otherwise to gain full advantage of this abstraction.